src/pages/trade/ui/positions/index.tsx

```diff
@@ -105,144 +136,169 @@ const Positions = observer(({ showInactive }: { showInactive: boolean }) => {
                     ))}
                   </Table.Tr>
                 ))}
-              {displayPositions
-                .filter(position => (showInactive ? true : !position.isWithdrawn))
-                .map(position => {
-                  return (
-                    <Table.Tr key={position.idString}>
-                      <Table.Td density='slim'>
-                        <div className='flex flex-col gap-4'>
-                          {position.orders
-                            .slice(0, position.isWithdrawn ? 1 : Infinity)
-                            .map((order, i) =>
-                              position.isOpened ? (
-                                <Text
-                                  as='div'
-                                  detail
-                                  color={
-                                    order.direction === 'Buy'
-                                      ? 'success.light'
-                                      : 'destructive.light'
-                                  }
-                                  key={i}
-                                >
-                                  {order.direction}
-                                </Text>
-                              ) : (
-                                <Text as='div' detail color='neutral.light' key={i}>
-                                  {stateToString(position.state)}
-                                </Text>
-                              ),
-                            )}
-                        </div>
-                      </Table.Td>
-                      <Table.Td density='slim'>
-                        <div className='flex flex-col gap-4'>
-                          {position.isWithdrawn ? (
-                            <Dash />
-                          ) : (
-                            position.orders.map((order, i) => (
-                              <ValueViewComponent
-                                key={i}
-                                valueView={
-                                  position.isClosed && i === 1 ? order.basePrice : order.amount
+              {orderBy(
+                displayPositions
+                  .filter(position => (showInactive ? true : !position.isWithdrawn))
+                  .map(position => ({
```

Comment:
suggestion (subjective): move the computation to the variables section of the component, maybe even to useMemo.

it might be a personal nitpick that prefer the vue/svelte style with logic/styles/render separation. In the end it's up to you

Reply:
if talking about nitpicks, can we limit the amount of lines of code per component? 😅

Reply:
I can see why you say that, I prefer in-line but in this case it's quite a lot so I agree with you. Updated it into a useMemo :+1:

Reply:
> if talking about nitpicks, can we limit the amount of lines of code per component? 😅

Could be good, what do you propose as the max lines?

Reply:
maybe 150, that's up for discussion